server/entity/item_behaviour.go

```diff
@@ -65,6 +67,30 @@ func (i *ItemBehaviour) Item() item.Stack {
 // Tick moves the entity, checks if it should be picked up by a nearby collector
 // or if it should merge with nearby item entities.
 func (i *ItemBehaviour) Tick(e *Ent) *Movement {
+	w := e.World()
+	pos := cube.PosFromVec3(e.Position())
+	blockPos := pos.Side(cube.FaceDown)
+
+	bl, ok := w.Block(blockPos).(block.Hopper)
+	if ok && !bl.Powered && bl.CollectCooldown <= 0 {
+		addedCount, err := bl.Inventory(w, pos).AddItem(i.i)
+		if err != nil {
+			// Check the added count is equal to the stack count, if not we create a new item entity
+			// with the updated stack count.
+			if addedCount != 0 {
+				w.AddEntity(NewItem(i.Item().Grow(addedCount-i.i.Count()), pos.Vec3Centre()))
+
+				_ = e.Close()
+				bl.CollectCooldown = 8
+				w.SetBlock(blockPos, bl, nil)
+			}
+			return i.passive.Tick(e)
+		}
```

Comment:
```suggestion
		if err != nil {
			if addedCount == 0 {
				return i.passive.Tick(e)
			}
			w.AddEntity(NewItem(i.i.Grow(addedCount-i.i.Count()), pos.Vec3Centre()))
		}
```

Reply:
Feel free to add a comment noting this is reached if only part of the item stack was collected into the hopper